src/test/java/com/google/devtools/build/lib/runtime/TestSummaryTest.java

```diff
@@ -518,6 +518,23 @@ public void testCollectingFailedDetails() throws Exception {
     verify(printer).print(find("ERROR.*cherry"));
   }
 
+  @Test
+  public void testCollectingPassedDetails() throws Exception {
```

Comment:
Can you add a test that has a mixture of FAILED, PASSED and ERROR cases and verify that all cases are printed? And update `testCollectingFailedDetails` to only check for failed/error cases.